Demo/src/main/java/com/paypal/android/ui/paypal/PayPalButtonsFragment.kt

```diff
@@ -0,0 +1,184 @@
+import androidx.fragment.app.Fragment
+import com.paypal.android.databinding.FragmentPaypalButtonsStylingBinding
+import com.paypal.android.ui.paymentbutton.*
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class PayPalButtonsFragment : Fragment() {
+
+    private val selectedButtonFundingType: ButtonFundingType
+        get() {
+            return ButtonFundingType.values()[
+                    binding.paymentButtonFundingSpinner.selectedItemPosition
+            ]
+        }
+
+    private val selectedColor: PaymentButtonColor?
+        get() = when (selectedButtonFundingType) {
+            ButtonFundingType.PAYPAL_CREDIT ->
+                binding.paymentButtonColorSpinner.selectedItem as? PayPalCreditButtonColor
+            else ->
+                binding.paymentButtonColorSpinner.selectedItem as? PayPalButtonColor
+        }
+
+    private val selectedLabel: PayPalButtonLabel?
+        get() = when (selectedButtonFundingType) {
+            ButtonFundingType.PAYPAL -> PayPalButtonLabel.valueOf(
+                binding.paymentButtonLabelSpinner.selectedItem.toString()
+            )
+            else -> null
+        }
+
+    private val selectedShape: PaymentButtonShape
+        get() = PaymentButtonShape.valueOf(
+            binding.paymentButtonShapeSpinner.selectedItem.toString()
+        )
+
+    private val selectedSize: PaymentButtonSize
+        get() = PaymentButtonSize.valueOf(
+            binding.paymentButtonSizeSpinner.selectedItem.toString()
+        )
+
+    private val binding: FragmentPaypalButtonsStylingBinding by lazy {
+        FragmentPaypalButtonsStylingBinding.inflate(layoutInflater)
+    }
+
+    private val itemSelectionListener = object : AdapterView.OnItemSelectedListener {
+        override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
+            if (parent?.id == binding.paymentButtonFundingSpinner.id) {
+                initPaymentButtonColorSpinner()
+                initPaymentButtonLabelSpinner()
+            }
+            showStyledButton()
+        }
+
+        override fun onNothingSelected(p0: AdapterView<*>?) {
+            //Do Nothing
+        }
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initFundingTypesSpinner()
+        initPaymentButtonColorSpinner()
+        initPaymentButtonLabelSpinner()
+        initPaymentButtonShapeSpinner()
+        initPaymentButtonSizeSpinner()
+    }
+
+    private fun initPaymentButtonSizeSpinner() {
+        binding.paymentButtonSizeSpinner.adapter = ArrayAdapter(
+            requireContext(),
+            android.R.layout.simple_list_item_1,
+            PaymentButtonSize.values()
+        )
```

Comment:
We could extract these `ArrayAdapter` factory methods into a single companion object to make the business logic easier to follow.

Reply:
@sshropshire I wouldn't want to put this in `companion object` because this is not static or util functionality, we can put in separate `val` now we're not re-using anywhere else in the code so I guess not needed